src/main/java/hudson/tasks/SMTPAuthentication.java

```diff
@@ -0,0 +1,40 @@
+package hudson.tasks;
+
+import hudson.Extension;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import hudson.util.Secret;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+public class SMTPAuthentication extends AbstractDescribableImpl<SMTPAuthentication> {
```

Comment:
Ideally it should use Credentials Plugin instead of manual implementation. But fine for now

Reply:
Hmmm I kinda agree, but this case can only accept username and password. I'd rather get this across the line.

it is long overdue 😱

Reply:
I've been making a try to see how to include credentials and it would be a more complex (and possibly breaking) change. I think it's worth to consider it for a separate PR.